### PR TITLE
Introduce script for phase 2 - aura tcl strat

### PR DESCRIPTION
**Pull request by petrovska-petro**
Tackle #701 

```
brownie run badger/aura_tcl_strategy
```

### CODE REVIEW

File: scripts/badger/aura_tcl_strategy.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+    # 2. organised splits for each asset
+    balance_bal = bal.balanceOf(vault)
+    balance_aura = aura.balanceOf(vault)
+    console.print(
+        f"[green] === Claimed rewards {balance_bal/1e18} BAL and {balance_aura/1e18} AURA === [/green]"
+    )
+
+    bal_swap_for_usdc = balance_bal * BAL_FOR_STABLES
+    aura_swap_for_usdc = balance_aura * AURA_FOR_STABLES
```
Comment (gosuto-inzasheru) [inline scripts/badger/aura_tcl_strategy.py:61]:
```suggestion
    bal_swap_for_usdc = int(balance_bal * BAL_FOR_STABLES)
    aura_swap_for_usdc = int(balance_aura * AURA_FOR_STABLES)
```
this prevents rounding problems which might arise in L70 and L72

File: scripts/badger/aura_tcl_strategy.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+    # 2.1 swap rewards for usdc
+    vault.cow.market_sell(bal, usdc, bal_swap_for_usdc, deadline=60 * 60 * 4, coef=COEF)
+    vault.cow.market_sell(
+        aura, usdc, aura_swap_for_usdc, deadline=60 * 60 * 4, coef=COEF
+    )
+
+    # 2.2 send to voter and deposit into aurabal/bauraBAL
+    aura.transfer(voter, balance_aura - aura_swap_for_usdc)
```
Comment (gosuto-inzasheru) [inline scripts/badger/aura_tcl_strategy.py:70]: could go for a direct lock here on behalf of no?
```suggestion
    safe.contract(r.aura.vlAURA).lock(voter, balance_aura - aura_swap_for_usdc)
```

File: scripts/badger/aura_tcl_strategy.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        bal.approve(wrapper, bal_to_deposit)
+        wrapper.deposit(
+            bal_to_deposit,  # uint256 _amount
+            wrapper_aurabal_out,  # uint256 _minOut
+            False,  # bool _lock
+            ZERO_ADDRESS,  # address _stakeAddress
+        )
+
+    # 3. dogfood in bauraBAL if in prod
+    vault_status = registry_v_2.productionVaultInfoByVault(bauraBal)[2]
```
Comment (gosuto-inzasheru) [inline scripts/badger/aura_tcl_strategy.py:113]: nice

File: scripts/badger/aura_tcl_strategy.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        # NOTE: should we just narrow the claim for graviAURA and modify `claim_all`?
+        # NOTE: short <2h window for signing as new cycles may be posted
+        vault.badger.claim_all()
+
+        # 3.1: send to tree for deficit coverage over time.
+        # will help together with the post-thresher trasnfer
+        bcvxcrv.transfer(r.badger_wallets.badgertree, bcvxcrv.balanceOf(vault))
+
+        # 3.2: send influence asset to voter
+        bvecvx.trasnfer(voter, bvecvx.balanceOf(vault))
```
Comment (gosuto-inzasheru) [inline scripts/badger/aura_tcl_strategy.py:133]:
```suggestion
        bvecvx.transfer(voter, bvecvx.balanceOf(vault))
```

**Review by gosuto-inzasheru — COMMENTED** [review]
tbh i would leave the tree claim part out completely; L120-L133.

all of those assets are autocompounding already anyway right? or is there a good reason to have them in the wallet instead? also introducing a 2h signing window on a 5/n will complicate things.

File: scripts/badger/aura_tcl_strategy.py (hunk 2 above)
Comment (petrovska-petro) [reply scripts/badger/aura_tcl_strategy.py]: good one, i pushed on latest commit

File: scripts/badger/aura_tcl_strategy.py (hunk 4 above)
Comment (petrovska-petro) [reply scripts/badger/aura_tcl_strategy.py]: removed this section, makes sense specially for the sensitivity of getting it sign under a narrow time window

**Review by gosuto-inzasheru — APPROVED** [review]
awesome! looking forward to capturing this in an avatar 😃

**Review by jalbrekt85 — APPROVED** [review]
lgtm

**Pull request merged**